File: app/src/main/java/com/hoymm/root/tictactoe2/GameEngine/GameHeaderFragment.java
package com.hoymm.root.tictactoe2.GameEngine;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.hoymm.root.tictactoe2.R;

/**
 * Created by Damian Muca - Kaizen (26.09.17)
 */

class GameHeaderFragment extends Fragment {
    private TextView circleTV, crossTV, drawsTV, whosTurnTV;
    private int circleScores = 0, crossScores = 0, drawsScores = 0;
    private GameFragsCommunication gameFragsCommunication;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.game_header, container, false);
    }


    public void onAttach(Context context){
        super.onAttach(context);
        try{
            gameFragsCommunication = (GameFragsCommunication)context;
        } catch(ClassCastException e){
            throw new ClassCastException(context.toString() + " must implement OnFragmentSendText");
        }
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        initTextViews();
        drawRandomMovementOrder();
        setValuesForComponents();
        super.onViewCreated(view, savedInstanceState);
    }

    private void initTextViews() {
        circleTV = getActivity().findViewById(R.id.circleScoreAmountID);
        crossTV = getActivity().findViewById(R.id.crossScoresAmountID);
        drawsTV = getActivity().findViewById(R.id.drawsScoresAmountID);
        whosTurnTV = getActivity().findViewById(R.id.turnValueID);
    }

    private void drawRandomMovementOrder() {

    }

    private void setValuesForComponents() {
        circleTV.setText(String.valueOf(circleScores));
        crossTV.setText(String.valueOf(crossScores));
        drawsTV.setText(String.valueOf(drawsScores));

        changeWhosTurnNowAlongWithDisplayingTextAndColor();
    }

    void changeWhosTurnNowAlongWithDisplayingTextAndColor() {
        whosTurnTV.setText(gameFragsCommunication.getWhosTurnNow());
        whosTurnTV.setTextColor(getWhosTurnTextColor());
    }

    private int getWhosTurnTextColor() {
        if (gameFragsCommunication.isCircleTurnNow())
            return ContextCompat.getColor(getContext(), R.color.circleBlue);
        return ContextCompat.getColor(getContext(), R.color.crossRed);
    }

    void increaseCircleScores(){
        circleTV.setText(++circleScores);
    }

    void increaseCrossScores(){
        crossTV.setText(++crossScores);
    }

    void increaseDrawsScores(){
        drawsTV.setText(++drawsScores);
    }
}

File: app/src/main/java/com/hoymm/root/tictactoe2/GameEngine/GameHeaderFragment.kt
package com.hoymm.root.tictactoe2.GameEngine

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

import com.hoymm.root.tictactoe2.R

/**
 * Created by Damian Muca - Kaizen (26.09.17)
 */

internal class GameHeaderFragment : Fragment() {
    private lateinit var whosTurnTV: TextView
    private var gameFragsCommunication: CurAppData? = null
    private lateinit var settingsIcon: ImageView

    private val whosTurnTextColor: Int get() =
            if (gameFragsCommunication!!.getWhoseTurnNow == Shape.circle)
                ContextCompat.getColor(context, R.color.circleBlue)
            else ContextCompat.getColor(context, R.color.crossRed)

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? =
            inflater!!.inflate(R.layout.game_header, container, false)

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        try {
            gameFragsCommunication = context as CurAppData
        } catch (e: ClassCastException) {
            throw ClassCastException(context!!.toString() + " must implement OnFragmentSendText")
        }

    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        initTextViews()
        drawRandomMovementOrder()
        setActionForSettingsButton()
        super.onViewCreated(view, savedInstanceState)
    }

    private fun initTextViews() {
        whosTurnTV = activity.findViewById(R.id.turnValueID)
        settingsIcon = activity.findViewById(R.id.settingsIcon)
    }

    private fun drawRandomMovementOrder() {

    }

    private fun setActionForSettingsButton(){
        settingsIcon.setOnClickListener{
            Toast.makeText(context, "Settings", Toast.LENGTH_SHORT).show()
        }
    }

    fun changeTurnToOpposiveTextView() {
        whosTurnTV.text = gameFragsCommunication!!.getWhoseTurnNow.toString()
        whosTurnTV.setTextColor(whosTurnTextColor)
    }

}
